iwyu_ast_util.cc

```diff
@@ -165,6 +165,18 @@ TemplateSpecializationKind GetTemplateSpecializationKind(const Decl* decl) {
   return clang::TSK_Undeclared;
 }
 
+const Type* GetNonAliasType(const Type* type) {
```

Comment:
Isn't this the equivalent of `QualType::getCanonicalType` or some other full desugaring?

Reply:
I don't want to lose `TemplateSpecializationType` (which is a sugar type with just a `RecordType` as the canonical type) along with its arguments.

Reply:
Ah, I missed the special case. Maybe `DesugarAliasTypes` to indicate that it's a form of desugaring after all?

A bit too early for cosmetic review, but I think that could help clarify what's going on.